src/main/java/parsing/Parser.java

```diff
@@ -118,8 +153,10 @@ public NodeGraph parse(final File file, NodeGraph graph) {
                                 out.flush();
                                 line2 = line2.substring(line2.indexOf('\t') + 1);
                                 line2 = line2.substring(line2.indexOf('\t') + 1);
-                                String nodeGenomes = line2.substring(0, line2.indexOf('\t'));
-                                addGenomes(gw, nodeGenomes);
```

Comment:
You replaced this addGenomes invocation with the new one. You didnt do that with the first invocation. Does this mean the first invocation could also have been replaced or is there no duplicate code.